cclunit-framework-source/doc/CCLUTMOCKING.md

```diff
@@ -0,0 +1,284 @@
+
+# CCL Unit Mocking
+The CCL Unit Test framework's mocking API is available for consumers to mock certain objects to better isolate unit tests from outside variability and provide more control over the scenarios under which unit tests are run.  The API provides ways to mock tables, variables, subroutines, scripts, etc.
+
+**\*CAUTION\*** **The CCL Unit Mocking framework should only be used in non-production environments.  Table mocking creates new tables against an Oracle instance for the lifetime of the test.  Because the DDL is generated in a dynamic way, it is possible through inappropriate use of the framework to affect the actual table.  Please only use the documented API.**
```

Comment:
> Should this be "an actual table" rather than "the actual table" ?
Yeah, that makes sense.

> There is also a chance that the mock tables will not be cleaned up (because CCL crashes mid-test). Should that statement be woven in as well?

Sure... though I might throw in a note that this is extremely rare (I've never, to my knowledge, had interactive CCL crash on me, only script servers) and that this is different than just simple errors (Some clients I've met weren't aware that CCL keeps processing after errors).

Reply:
Hit CTRL-C twice in the middle of a maven build. The current backend session will be terminated and no cleanup will occur.

I have lost track of the example, but at one time I knew of an innocent construct that would cause CCL to abort. I imagine there still is one.

Reply:
Fair enough.  I actually just pushed up my next batch of changes which has my proposed wording for this.  Feel free to look it over and let me know if you want me to change any of it.